pkgs/os-specific/bsd/openbsd/default.nix

```
@@ -0,0 +1,894 @@
+  stat = mkDerivation {
+    path = "usr.bin/stat";
+    version = "7.2";
+    sha256 = "18nqwlndfc34qbbgqx5nffil37jfq9aw663ippasfxd2hlyc106x";
+    nativeBuildInputs = with buildPackages.openbsd; [
+      bsdSetupHook openbsdSetupHook
+      mandoc groff rsync bmake
+    ];
+    buildPhase = ''
+      ${bmake}/bin/bmake
+      '';
+  };
+
+  # stat isn't in POSIX, and OpenBSD stat supports a completely
+  # different range of flags than GNU stat, so including it in PATH
+  # breaks stdenv.  Work around that with a hook that will point
+  # OpenBSD's build system and OpenBSD stat without including it in
+  # PATH.
+  statHook = makeSetupHook {
+    name = "openbsd-stat-hook";
+  } (writeText "openbsd-stat-hook-impl" ''
+    makeFlagsArray+=(TOOL_STAT=${self.stat}/bin/stat)
+  '');
+
+  tsort = mkDerivation {
+    path = "usr.bin/tsort";
+    version = "7.2";
+    sha256 = "1dqvf9gin29nnq3c4byxc7lfd062pg7m84843zdy6n0z63hnnwiq";
+    nativeBuildInputs = with buildPackages.openbsd; [
+      bsdSetupHook openbsdSetupHook
+      mandoc groff rsync
+    ];
+  };
+
+  lorder = mkDerivation {
+    path = "usr.bin/lorder";
+    version = "7.2";
+    sha256 = "0rjf9blihhm0n699vr2bg88m4yjhkbxh6fxliaay3wxkgnydjwn2";
+    nativeBuildInputs = with buildPackages.openbsd; [
+      bsdSetupHook openbsdSetupHook
+      mandoc groff rsync
+    ];
+  };
+
+  ##
+  ## END BOOTSTRAPPING
+  ##
+
+  ##
+  ## START COMMAND LINE TOOLS
+  ##
+  make = mkDerivation {
+    path = "usr.bin/make";
+    sha256 = "sha256-hhxQ88GWjyKlqODv66uYAG8mtjSFp2cpqqaOVXdEMls=";
+    version = "7.2";
+
+    buildInputs = with self; [ include ];
+
+   postPatch = ''
+     substituteInPlace $BSDSRCDIR/share/mk/bsd.lib.mk \
+       --replace '-o ''${LIBOWN}' "" \
+       --replace '-g ''${LIBGRP}' "" \
+       --replace '-o ''${DEBUGOWN}' "" \
+       --replace '-g ''${DEBUGGRP}' ""
+     substituteInPlace $BSDSRCDIR/share/mk/bsd.man.mk \
+       --replace '-o ''${MANOWN}' "" \
+       --replace '-g ''${MANGRP}' ""
+     substituteInPlace $BSDSRCDIR/share/mk/bsd.prog.mk \
+       --replace '-o ''${BINOWN}' "" \
+       --replace '-g ''${BINGRP}' "" \
+       --replace '-o ''${RUMPBINOWN}' "" \
+       --replace '-g ''${RUMPBINGRP}' "" \
+       --replace '-o ''${DEBUGOWN}' "" \
+       --replace '-g ''${DEBUGGRP}' ""
+
+      # make needs this to pick up our sys make files
+      export NIX_CFLAGS_COMPILE+=" -D_PATH_DEFSYSPATH=\"$out/share/mk\""
+
+      substituteInPlace $BSDSRCDIR/share/mk/bsd.lib.mk \
+        --replace '_INSTRANLIB=''${empty(PRESERVE):?-a "''${RANLIB} -t":}' '_INSTRANLIB='
+    '' + lib.optionalString stdenv.isDarwin ''
+      substituteInPlace $BSDSRCDIR/share/mk/bsd.sys.mk \
+        --replace '-Wl,--fatal-warnings' "" \
+        --replace '-Wl,--warn-shared-textrel' ""
+    '';
+
+    nativeBuildInputs = with buildPackages.openbsd; [
+      bsdSetupHook openbsdSetupHook rsync
+    ];
+
+    buildPhase = ''
+      ${bmake}/bin/bmake
```

This looks like we would want

```suggestion
      bmake
```
and nativeBuildInputs